moralization/transformers_data_handler.py

```diff
@@ -94,3 +101,67 @@ def init_tokenizer(self, model_name="bert-base-cased", kwargs=None):
             raise ValueError(
                 "Please use a different model that provices a fast tokenizer"
             )
+
+    def tokenize(self, wordlist: list = None):
+        if wordlist is None:
+            wordlist = self.token_list
+        wordlist = self._check_is_nested(wordlist)
+        self.inputs = self.tokenizer(
+            wordlist, truncation=True, is_split_into_words=True
+        )
+
+    def _check_is_nested(self, list_to_check: list) -> list:
+        # make sure that data is a nested list,
+        # otherwise add a layer
+        # do we need this?
+        # maybe enough to check if list_to_check[0] is a list?
+        # does it cost us to iterate over all the data?
+        list_to_check = (
+            [list_to_check] if not isinstance(list_to_check[0], list) else list_to_check
+        )
+        list_to_check = [[i] if not isinstance(i, list) else i for i in list_to_check]
```

Comment:
may be worth checking if the list is already nested first before making a new list? (it means checking twice when the list is not nested, but I would guess making a new list is the expensive part here)
```suggestion
        if not all(isinstance(i, list) for i in list_to_check):
            list_to_check = [[i] if not isinstance(i, list) else i for i in list_to_check]
```